### ISSUE TITLE
[layers] Add permutation-equivariant DeepSets block

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#5659]
## Summary

Add a single **permutation-equivariant linear layer** (Deep Sets) as a first-class
Odyssey `Module`, so set-structured inputs can be processed with a primitive whose
output is equivariant to permutations of the set elements. This is the reusable
Odyssey primitive that the downstream PC-vs-BP ARCH-14 ablation
(`mvillmow/Random#64`) builds on.

## Layer

Zaheer et al. 2017, "Deep Sets" (NeurIPS 2017, arXiv:1703.06114), equivariant
form — SUM pool, ReLU activation:

```
y_i = relu( x_i @ Lambda  +  (sum_j x_j) @ Gamma  +  b )
```

Batch-first `[batch, set_size, dim]`: the set axis (axis 1) plays the role the
sequence axis plays in the recurrent layers; pooling reduces over it. The pooled
term `sum_j x_j` is permutation-INVARIANT and is broadcast identically onto every
set element — that invariant + shared-per-element structure is exactly the
necessary-and-sufficient condition for equivariance derived in the paper
(Sec. "Permutation Equivariance"). `Lambda`, `Gamma` are `(dim, out)`; `b` is
`(out,)`.

## Scope

- `src/odyssey/core/layers/deepsets.mojo` — `DeepSetsEquivariant` struct
  (`Module` trait; float32 default; `Self.dtype` compile-time contract).
- Export from `odyssey.core.layers` (docstring Components line + import line).
- Unit tests under `tests/odyssey/core/layers/` + a torch parity generator under
  `parity_refs/`.
- README primitive-test table row + `scripts/run_primitive_test.sh` registration.

## Acceptance criteria

- [x] `DeepSetsEquivariant` implements the `Module` trait
      (`forward`/`parameters`/`train`/`eval`), batch-first `[batch, set, dim]`.
- [x] Forward matches a PyTorch reference to 1e-5 (ramp-seeded, f64 ground truth).
- [x] Dedicated permutation-equivariance test: a nontrivial set-axis permutation
      of the input permutes the output rows identically (to 1e-9).
- [x] Importable via `from odyssey.core.layers import DeepSetsEquivariant`
      (package-path import test).
- [x] `mojo format` clean; `--Werror` compile clean; ruff + markdownlint clean;
      pre-commit passes on touched files.
- [x] No CHANGELOG / lockfile churn; only the owned shared-file lines touched.

## Reference

Zaheer, M., Kottur, S., Ravanbakhsh, S., Poczos, B., Salakhutdinov, R., Smola, A.
(2017). *Deep Sets.* NeurIPS 2017. arXiv:1703.06114.

## Tracking

`mvillmow/Random#64` (ARCH-14, 1-layer permutation-equivariant model). This issue
covers only the Odyssey layer primitive; the PC/BP sweep wiring and the
optimizer×clip×LR matrix run remain in the tracking issue.
